File: shared_resources/src/main/java/com/thewizrd/shared_resources/BatteryStatus.java
package com.thewizrd.shared_resources;

public class BatteryStatus {
    public int batteryLevel;
    public boolean isCharging;

    public BatteryStatus(int batteryLevel, boolean isCharging) {
        this.batteryLevel = batteryLevel;
        this.isCharging = isCharging;
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/actions/BatteryStatus.kt
package com.thewizrd.shared_resources.actions

class BatteryStatus(var batteryLevel: Int, var isCharging: Boolean) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is BatteryStatus) return false

        if (batteryLevel != other.batteryLevel) return false
        if (isCharging != other.isCharging) return false

        return true
    }

    override fun hashCode(): Int {
        var result = batteryLevel
        result = 31 * result + isCharging.hashCode()
        return result
    }
}